experimental/ModelBuilder/test/CMakeLists.txt

```
@@ -0,0 +1,276 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+iree_add_all_subdirs()
+
+file(GLOB _GLOB_X_CPP LIST_DIRECTORIES false RELATIVE ${CMAKE_CURRENT_SOURCE_DIR} CONFIGURE_DEPENDS *.cpp)
+file(GLOB _GLOB_BENCHX LIST_DIRECTORIES false RELATIVE ${CMAKE_CURRENT_SOURCE_DIR} CONFIGURE_DEPENDS Bench*)
+list(REMOVE_ITEM _GLOB_X_CPP ${_GLOB_BENCHX})
+iree_lit_test_suite(
+  NAME
+    lit
+  SRCS
+    "${_GLOB_X_CPP}"
+  DATA
+    test-dot-prod
+    test-matmul-vulkan
+    test-mnist-jit
+    test-simple-jit
+    test-simple-jit-vulkan
+    test-simple-mlir
+    test-vec-to-gpu
+#   test-vector-transfers-jit
+    iree::tools::IreeFileCheck
+)
+
+iree_cc_binary(
+  NAME
+    test-dot-prod
+  OUT
+    test-dot-prod
+  SRCS
+    "TestDotProdJIT.cpp"
+  DEPS
+    LLVMSupport
+    MLIRAllDialects
+    MLIREDSC
+    MLIRIR
+    MLIRSCFTransforms
+    experimental::ModelBuilder
+    experimental::ModelBuilder::ModelRunner
+)
+
+# TODO(thomasraoux): Enable runtime-support.so build.
+#iree_cc_binary(
+#  NAME
+#    test-vector-transfers-jit
+#  OUT
+#    test-vector-transfers-jit
+#  SRCS
+#    "TestVectorTransfersJIT.cpp"
+#  DEPS
+#    ::runtime-support.so
+#    LLVMSupport
+#    MLIRAllDialects
+#    MLIREDSC
+#    MLIRIR
+#    MLIRSCFTransforms
+#    experimental::ModelBuilder
+#    experimental::ModelBuilder::ModelRunner
+#)
+
+iree_cc_binary(
+  NAME
+    test-mnist-jit
+  OUT
+    test-mnist-jit
+  SRCS
+    "TestMNISTJIT.cpp"
+  DEPS
+    MLIRAllDialects
```

As commented before, it might be worth to check if
```suggestion
    init_mlir_passes_and_dialects
```
is an alternative. We could further refactor `init_mlir_passes_and_dialects`, to split into separate dialects and passes targets if required.

---

Obsolete as dependencies on IREE should be avoided.